contributed_definitions/nyaml/NXmanipulator.yaml

```diff
@@ -13,32 +13,124 @@ NXmanipulator(NXobject):
   type(NX_CHAR):
     doc: |
       Type of manipulator, Hexapod, Rod, etc.
-  cryocoolant(NX_BOOLEAN):
+  cryostat(NXactuator):
     doc: |
-      Is cryocoolant flowing through the manipulator?
-  cryostat_temperature(NX_FLOAT):
-    unit: NX_TEMPERATURE
+      Cryostat for cooling the sample.
+    physical_quantity:
+      enumeration: [temperature]
+    (NXpid):
+      setpoint(NX_FLOAT):
+        unit: NX_TEMPERATURE
+        doc: |
+          In case of a fixed or averaged cooling temperature, this is the scalar temperature setpoint.
+          It can also be a 1D array of temperature setpoints (without time stamps).
+      setpoint_log(NXlog):
+        value(NX_FLOAT):
+          unit: NX_TEMPERATURE
+          doc: |
+            In the case of an experiment in which the temperature is changed and the setpoints are
+            recorded with time stamps, this is an array of length m of temperature setpoints.
+  temperature_sensor(NXsensor):
     doc: |
-      Temperature of the cryostat (coldest point)
-  heater_power(NX_FLOAT):
-    unit: NX_POWER
+      Temperature sensor measuring the sample temperature.
+    measurement:
+      enumeration: [temperature]
+    value(NX_FLOAT):
+      unit: NX_TEMPERATURE
+      doc: |
+        In case of a single or averaged temperature measurement, this is the scalar temperature measured
+        by the sample temperature sensor. It can also be a 1D array of measured temperatures
+        (without time stamps).
+    value_log(NXlog):
```

Comment:
The constructs with the NXlogs here point to an issue which I also struggled with namely how to detail correlations between fields, should be discussed further in the TF but okay to me here for now